lib/sshutils/scp/scp.go

```diff
@@ -478,14 +483,13 @@ func (cmd *command) processCommand(ch io.ReadWriter, st *state, b byte, line str
 func (cmd *command) receiveFile(st *state, fc newFileCmd, ch io.ReadWriter) error {
 	cmd.log.Debugf("scp.receiveFile(%v): %v", cmd.Flags.Target, fc.Name)
 
-	// if the destination path is a folder, we should save the file to that folder, but
-	// only if 'recursive' is set
-
-	path := cmd.Flags.Target[0]
-	if cmd.Flags.Recursive || cmd.FileSystem.IsDir(path) {
-		path = st.makePath(fc.Name)
+	// Unless target specifies a file, use the file name from the command
+	filename := fc.Name
+	if !cmd.Flags.Recursive && !cmd.FileSystem.IsDir(cmd.Flags.Target[0]) {
```

Comment:
Is `cmd.Flags.Target` guaranteed to have at least 1 element here?

Reply:
Yes, I've updated the client side of SCP to send files in bulk for source mode and the caller makes sure that at least 1 target item is always set.